exercises/react/example.cr

```diff
@@ -0,0 +1,71 @@
+alias CellValue = Int32
+
+private class Cell
+  @dependents = [] of ComputeCell
+
+  def initialize(initial : CellValue)
+    @value = initial
+  end
+
+  getter :value
+
+  protected getter :dependents
+  protected def propagate; _propagate; end
```

Comment:
The behaviour is the same as in Ruby, so it is to be expected.

If you attempt to run the above in Ruby, you will succeed if you are just doing `SomeThing.new([Thing.new, Thing.new]).hi_friends` But as soon as you try `SomeThing.new([Thing.new, GoodThing.new]).hi_friends` you will get at run-time: ```protected_methods.rb:13:in `each': protected method `hi' called for #<GoodThing:0x000000024d4ea8 @friends=[]> (NoMethodError)```

You have to do the protected -> private delegation in Ruby as well. So the two languages are doing it the same way. This tells me that this is expected. And now we see that Crystal protects us at compile-time from problems that only happen at run-time in Ruby.